engine/src/application.cpp

```diff
@@ -20,26 +21,33 @@ namespace Bald {
     }
 
     void Application::Run() {
-        m_EventManager->Subscribe<WindowClosedEvent>(HandleType::SYNC, [&](const WindowClosedEvent&) {
-            CORE_LOG_TRACE("Window Closed Event!");
-            glfwSetWindowShouldClose(m_Window->GetWindow(), true);
-            m_Running = false;
-        });
+
         while(m_Running) {
             m_Window->Clear();
 
-            // TODO: Update layer stack here
-            m_Window->Update();
-            Input::InputManager::Update(); // TODO: This should probably be called on layer update ~Blinku
+            for(size_t i = 0; i < m_LayerStack.GetSize(); ++i) {
+                m_LayerStack[i]->OnUpdate();
+            }
+
+            for(size_t i = m_LayerStack.GetSize(); i != 0; --i) {
+                m_LayerStack[i - 1]->RunEvents();
+            }
+
             m_EventManager->Flush();
+
             EventManager::ClearEventQueue();
+
+            Input::InputManager::Update();
+
+            m_Window->Update();
```

Comment:
Probably updates should be called before Running events.

Reply:
Calling them before messes everything up. This needs further investigation.